src/tar_header_parser.e

```diff
@@ -33,18 +33,18 @@ feature -- Result query
 		require
 			completely_parsed: parsing_finished
 		do
-			Result := current_header
+			Result := last_parsed_header
 		end
 
 feature {NONE} -- Implementation
 
-	current_header: detachable TAR_HEADER
+	last_parsed_header: detachable TAR_HEADER
 			-- The current header (based on the parsed blocks)
 			-- Void in case of parse failures
 
 feature {NONE} -- Utilities
 
-	parse_string (block: MANAGED_POINTER; pos, length: INTEGER): STRING
+	next_block_string (block: MANAGED_POINTER; pos, length: INTEGER): STRING
```

Comment:
in Eiffel, we use name for feature returning value  (i.e function or attribute) and verb for procedure.

Maybe "next_block_string" is wrong .. it should be "block_string" or "parsed_string" ...